daemon/config/config.go

```diff
@@ -244,7 +245,10 @@ func Reload(configFile string, flags *pflag.FlagSet, reload func(*Config)) error
 	logrus.Infof("Got signal to reload configuration, reloading from: %s", configFile)
 	newConfig, err := getConflictFreeConfiguration(configFile, flags)
 	if err != nil {
-		return err
+		if flags.Changed("config-file") || !os.IsNotExist(err) {
+			return fmt.Errorf("unable to configure the Docker daemon with file %s: %v", configFile, err)
+		}
+		return nil
```

Comment:
If a config file is not specified and does not exist, I believe the configuration should still be reloaded. It looks like this simply returns success in this case without reloading the configuration (i.e. resetting whatever used to be in the config that has now been deleted, ensuring flags are obeyed, etc).

Reply:
Hey @dsheets ,
Thank you for the review but I probably miss something. 😕 
If `/etc/docker/daemon.json` (which is the default configuration file) doesn't exist or was removed, then there are no changes available, so what is needed to be reloaded.

😀

Reply:
From #35114:

1. Start `dockerd` without a config file (config 0)
2. Create `/etc/docker/daemon.json` and populate it with some new configuration
3. Send `SIGHUP` to `dockerd` and notice the new configuration is picked up (config 1)
4. Delete `/etc/docker/daemon.json`
5. Send `SIGHUP` to `dockerd` and notice that the configuration does not change (config 1)

I believe this reproduction will have the same behavior with your patch. In particular, step 4 changes the configuration by removing the configuration file and, after step 5, the configuration should revert to "config 0" as was used in step 1.

Reply:
Oh I got it. There should be a default configuration if the `/etc/docker/daemon.json` doesn't exist and no other configuration file was given.
Right?

Reply:
Yes, that's correct.

Reply:
Thanks! I am going to work on this change.

Reply:
Yes, we should probably have something like;

- start daemon
- create `daemon.json` and reload
- check the configuration was updated
- remove `daemon.json` and reload
- check the configuration was reverted to the original settings

One interesting bit would be; what should happen to settings that are not live-reloadable but were present in the `daemon.json`? (perhaps we already take those into account)

Reply:
I believe that fields that are not reloadable are simply ignored. See <https://github.com/moby/moby/blob/master/daemon/reload.go#L13>.